social-network-data-migration-cli/src/main/java/pl/edu/agh/socialnetworkdatamigration/cli/SocialNetworkDataMigrationCLI.java

```diff
@@ -1,88 +1,153 @@
+
+            if (cmd.hasOption('i')) {
+                schemaMapping = new InteractiveCSVMappingCreator(dataPath, hasHeaders).createInteractively();
+            } else {
+                String mappingsPath = cmd.getOptionValue("mapping");
+                String jsonStr = Files.readString(Path.of(mappingsPath));
+                schemaMapping = new CSVMappingLoader(dataPath, hasHeaders).loadFromJson(jsonStr);
+            }
 
-    private static void SQLmigration(String[] args) throws Exception {
-        String configPath = args[0];
-        String mappingsPath = args[1];
+            if (cmd.hasOption("merge")) {
+                migrationStrategy = new CsvMergingStrategy(dataPath, fieldTerminator, hasHeaders);
+            } else {
+                migrationStrategy = new CsvAddingStrategy(dataPath, fieldTerminator, hasHeaders);
+            }
+            try (Driver neo4jDriver = GraphDatabase.driver("neo4j://" + neo4jHost, AuthTokens.basic(neo4jUser, neo4jPassword));
+                 Migrator<CSVSchemaMapping> migrator = new Migrator<>(neo4jDriver, migrationStrategy, cmd.hasOption("dry-run"))) {
+                migrator.migrateData(schemaMapping);
+            }
+        } else {
+            String postgresHost = properties.getProperty("postgresHost");
+            String postgresDB = properties.getProperty("postgresDB");
+            String postgresUser = properties.getProperty("postgresUser");
+            String postgresPassword = properties.getProperty("postgresPassword");
+            SchemaMetaData schemaMetaData = new SchemaMetaData(postgresHost, postgresDB, postgresUser, postgresPassword);
+            SQLSchemaMapping schemaMapping;
+            PostgresStrategy migrationStrategy;
 
-        var mappingLoader = new SQLMappingLoader();
-        String jsonStr = Files.readString(Path.of(mappingsPath));
-        var schemaMapping = mappingLoader.loadFromJson(jsonStr);
+            if (cmd.hasOption('i')) {
+                schemaMapping = new InteractiveSQLMappingCreator(schemaMetaData.getDatabaseInfo()).createInteractively();
+            } else {
+                String mappingsPath = cmd.getOptionValue("mapping");
+                String jsonStr = Files.readString(Path.of(mappingsPath));
+                schemaMapping = new SQLMappingLoader().loadFromJson(jsonStr);
+            }
 
-        try (var migrator = new PostgresMigrator(configPath)) {
-            migrator.migrateData(schemaMapping);
+            if (cmd.hasOption("merge")) {
+                migrationStrategy = new PostgresMergingStrategy(schemaMetaData);
+            } else {
+                migrationStrategy = new PostgresAddingStrategy(schemaMetaData);
+            }
+
+            try (Driver neo4jDriver = GraphDatabase.driver("neo4j://" + neo4jHost, AuthTokens.basic(neo4jUser, neo4jPassword));
+                 Migrator<SQLSchemaMapping> migrator = new Migrator<>(neo4jDriver, migrationStrategy, cmd.hasOption("dry-run"))) {
```

Comment:
`dry-run` powinno być przekazane jako parametr metody `migrateData`, a nie w konstruktorze migratora

Reply:
Zgadza się, zresztą zgodnie z nazwą powinno być to cechą wywołania.